src/java/htsjdk/samtools/util/Locatable.java

```diff
@@ -22,4 +22,68 @@
      * @return 1-based closed-ended position, undefined if getContig() == null
      */
     int getEnd();
+
+
+    /**
+     * @return number of bases covered by this interval
+     */
+    default int size() {
+        return CoordMath.getLength(getStart(), getEnd());
+    }
+
+    /**
+     * Determines whether this interval overlaps the provided locatable.
+     *
+     * @param other interval to check
+     * @return true if this interval overlaps other, otherwise false
+     */
+    default boolean overlaps(Locatable other) {
+        return withinDistanceOf(other, 0);
```

Comment:
What about adjacent (but not overlapping) intervals? Are they not "within 0 distance" of each other?